src/perf_tool_backend.ml

```diff
@@ -63,6 +63,38 @@ let perf_fork_exec ?env ~prog ~argv () =
   | `In_the_parent pid -> pid
 ;;
 
+let max_sampling_frequency () =
+  In_channel.read_all "/proc/sys/kernel/perf_event_max_sample_rate"
+  |> String.rstrip (* Strip off newline *)
+  |> Int.of_string
+;;
+
+(* This checks if pdcm flag is present in lscpu. This is necessary for LBR to work.
+   Although I couldn't ascertain that it is also sufficient. *)
+let supports_lbr () =
+  let open Deferred.Or_error in
+  let%bind.Deferred lscpu_result = Process.run ~prog:"lscpu" ~args:[] () in
```

Comment:
We can't rely on `lscpu` always being present, it is an optional package that doesn't get installed alongside `perf`. However, the data you need here is also present in `/proc/cpuinfo` in almost the same format.

Reply:
I refactored to just check if all CPUs in /proc/cpuinfo contain the flag pdcm. It's also now located in `perf_capabilities.ml`.